packages/abtest/src/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import { CONTROL } from './constants';
+import {
+	getAndSetGroup,
+	getCachedGroup,
+	isActive,
+	recordABTestEvent,
+} from './utils';
+
+/**
+ * ABTest component.
+ *
+ * Use to randomly display one of two components to control and experiment groups.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.name Name of the experiment.
+ * @param {Object} props.control React component to display to control group.
+ * @param {Object} props.experiment React component to display to experiment group.
+ * @param {Function|null} props.onComplete Optional. Callback triggered once A/B Test has completed setup.
+ * @param {number} props.size Optional. Size of experiment group.
+ * @param {number} props.start Optional. A/B Test start timestamp.
+ * @param {number} props.end Optional. A/B Test end timestamp.
+ */
+const ABTest = ( {
+	name,
+	control,
+	experiment,
+	onComplete,
+	size = 50,
+	start = 0,
+	end = Infinity,
+} ) => {
+	const [ group, setGroup ] = useState( CONTROL );
+	const [ isFetching, setIsFetching ] = useState( true );
+	const active = isActive( start, end );
+	const handleComplete = useCallback( () => {
+		setIsFetching( false );
+		if ( onComplete ) {
+			onComplete();
+		}
+	}, [ onComplete ] );
+
+	const setABTestOption = useDispatch( OPTIONS_STORE_NAME ).updateOptions;
+	const getABTestOption = useSelect(
+		( select ) => select( OPTIONS_STORE_NAME ).getOption
+	);
+
+	useEffect( () => {
+		if ( active ) {
+			const cachedGroup = getCachedGroup( name );
+			if ( cachedGroup ) {
+				setGroup( cachedGroup );
+				recordABTestEvent( name, cachedGroup, 'from_cache', 'serve' );
+			} else {
+				const newGroup = getAndSetGroup(
+					name,
+					size,
+					getABTestOption,
+					setABTestOption
+				);
+				setGroup( newGroup );
+				recordABTestEvent( name, newGroup, 'from_store', 'serve' );
+			}
+		}
+		handleComplete();
```

Comment:
If a test has expired past an end date, `handleComplete` will be called every time. Is that the intention? The description `A callback fired after A/B Test selects group` maybe suggests otherwise.

Reply:
Ah. Good catch! 

I think it's best not to fire the cb when the test is not active. I've changed this in https://github.com/woocommerce/woocommerce-admin/pull/6536/commits/fb2796bbc7f0a4100a16e3cf9c2bf16c5a6fce9b